### PR TITLE
Adds search unordered items in parallel example

### PR DESCRIPTION
**Pull request by j-haj**
cc #330

### CODE REVIEW

**Comment by j-haj** [comment]
Didn't mean for c3f4a42 do be included in this.. I will rebase and push and update

File: src/concurrency.md [hunk 1]
```diff
@@ -72,6 +73,46 @@ fn main() {
...
+use rayon::prelude::*;
+
+fn main() {
+    let v = vec![6, 2, 1, 9, 3, 8, 11];
+
+    let f1 = v.par_iter().find_any(|&&x| x == 9);
+    let f2 = v.par_iter().find_any(|&&x| x % 2 == 0 && x > 6);
+    let f3 = v.par_iter().find_any(|&&x| x > 8);
+
+    if let Some(found_9) = f1 {
```
Comment (budziq) [inline src/concurrency.md]: How about writing it more concisely as
```rust
    assert_eq!(f1, Some(&9));
    assert_eq!(f2, Some(&8));
    assert!(f3 > Some(&8))
```

File: src/concurrency.md [hunk 2]
```diff
@@ -72,6 +73,46 @@ fn main() {
 }
 ```
 
+[ex-rayon-parallel-search]: #ex-rayon-parallel-search
+<a name="ex-rayon-parallel-search"></a>
+## Search unordered items in parallel
```
Comment (budziq) [inline src/concurrency.md]: Hmm we might want to change the example title to something along the lines of
"Search items with given predicate in parallel". Originally the "unordered" part was to emphasize that we are not doing binary search but I think that mentioning the arbitrary predicate is a stronger usecase.

File: src/concurrency.md (hunk 1 above)
Comment (j-haj) [reply src/concurrency.md]: This is much cleaner. It did not occur to me to convert the expected value into an `optional`. Very clever! 👍

**Comment by j-haj** [comment]
Great suggestions! I'll push some updates later this evening

**Comment by j-haj** [comment]
I slightly modified the suggested title to "Search items *using* given predicate in parallel" to make it clear the predicate is associated with the search rather than the items.

**Comment by budziq** [comment]
Nicely done @j-haj !

**Pull request merged**